widget/embedded/src/hooks/useBootstrap/useBootstrap.ts

```diff
@@ -0,0 +1,67 @@
+import type { WalletType } from '../..';
+
+import { useQueueManager } from '@rango-dev/queue-manager-rango-preset';
+import { splitWalletNetwork } from '@rango-dev/wallets-shared';
+import { isEvmBlockchain } from 'rango-sdk';
+import { useContext, useEffect, useState } from 'react';
+
+import { useWallets } from '../..';
+import { WidgetContext } from '../../containers/Wallets';
+import { globalFont } from '../../globalStyles';
+import { useAppStore } from '../../store/AppStore';
+import { useNotificationStore } from '../../store/notification';
+import { tabManager } from '../../store/ui';
+import { useFetchApiConfig } from '../useFetchApiConfig';
+import { useForceAutoConnect } from '../useForceAutoConnect';
+
+export function useBootstrap() {
+  globalFont();
+  useForceAutoConnect();
+  const blockchains = useAppStore().blockchains();
+  const { canSwitchNetworkTo } = useWallets();
+  const [lastConnectedWalletWithNetwork, setLastConnectedWalletWithNetwork] =
+    useState('');
+  const [disconnectedWallet, setDisconnectedWallet] = useState<WalletType>();
+  const widgetContext = useContext(WidgetContext);
+
+  const evmChains = blockchains.filter(isEvmBlockchain);
+
+  const { fetchApiConfig } = useFetchApiConfig();
+
+  useQueueManager({
+    lastConnectedWallet: lastConnectedWalletWithNetwork,
+    clearDisconnectedWallet: () => {
+      setDisconnectedWallet(undefined);
+    },
+    disconnectedWallet: disconnectedWallet,
+    evmChains,
+    canSwitchNetworkTo,
+  });
+
+  useEffect(() => {
+    tabManager.init();
+
+    if (!useNotificationStore.persist.hasHydrated()) {
+      void useNotificationStore.persist.rehydrate();
+    }
+
+    /**
+     * Add handlers to the widget context for connecting and disconnecting wallets.
+     * We pass these handlers to the wallet provider to keep track of the last connected and last disconnected wallets.
+     * The 'queue-manager' uses these details to spot any changes in the connection status of the required wallet during the swap process.
+     */
+    widgetContext.onConnectWallet(setLastConnectedWalletWithNetwork);
```

Comment:
I'm concerned about the current approach. We've placed `onConnectWallet` and `onDisconnectWallet` within `WidgetContext`, making them accessible everywhere. My concern is if someone else attempts to do the same (`widgetContext.onDisconnectWallet(  somethingElseFunction )`), it could lead to conflicts. I guess the last one defined and called would override the context so it can not be reliable and potentially unsafe .
If my understanding is accurate, How we can guard these handlers safely to not gone?

Reply:
as we talked offline, we decided to handle this somehow in future,